README.md

```diff
@@ -153,6 +153,14 @@ usage is clear you can use `U`, `V` etc. for any following generic.
 If the usage is not obvious, you should use a more descriptive name. The same naming rules as for
 classes will apply then.
 
+### Prefixing
+
+#### Types and Interfaces
+
+Do not prefix types with a `T` or interfaces with an `I`. For example `new Array<TData>();` there
+is no need to prefix them with a `T`. It's even better to give the type a descriptive name like
+`new Array<Person>();`.
```

Comment:
This example is not clear, because this is a use case of a generic instead of an definition of a generic.

maybe something like:

Don't
```
class Response<TData> {
  public data: TData;
}

```

Do
```
class Response<Data> {
  public data: Data;
}

```

Reply:
I'd say don't do either and just use `T`

```ts
class Response<T> {
  public data: T;
}
```

naming it `Data` doesn't help in anyway in understanding what the intent of the generic is. Personally, it confuses me because I immediately think that there's a `Data` type or interface around somewhere.

Reply:
Agreed with @psimk here, if you want to add content to your type param I suggest you make it extend a named label, this has the added benefit that it puts (optional) constraints on the type. 

```typescript
type ResponseData = Record<string, unknown>

class Response<T extends ResponseData> {
  public data: T;
}
```

Reply:
I like the example of Nathan. You have the advance of both: it's clear that the type of `data` is a generic, and in the definition of `T` you see that it is used for the `ResponseData`